File: Practice/17/Python/17th.py
count = {}
drop_round = {}
for i in range(37):
    count[i] = 0
    drop_round[i] = 0

high_number = 0;
Round = 0;
reds = 0;
blacks = 0
while (True):
    x = int(input())
    if x < 0:
        break
    Round = Round + 1
    count[x] = count[x] + 1
    drop_round[x] = Round
    
    high_number = max(count.values())
    for i in range(37):
        if count[i] == high_number:
            print(i, end = ' ')
    print()

    for i in range(37):
        if (drop_round[i] == 0) or (Round-drop_round[i]>12):
            print(i, end = ' ')
    print()

    if x != 0:
        if (x <= 10) or (x > 18 and x <= 28):
            if x % 2 == 1: reds += 1
            else:             blacks += 1
        else:
            if x % 2 == 0: reds += 1
            else:             blacks += 1

    print(reds,blacks,'\n')

File: Practice/17/C++/17th/17th/17th.cpp
#include <iostream>
int main() {
	const int NUM = 37;
	const int k = 12;
	int x, blacks = 0, reds = 0;
	int round_drop[NUM];
	int count[NUM];
	for (int i = 0; i < NUM; i++)
	{
		count[i] = 0;
		round_drop[i] = 0;
	}
	int round=0;
	int high_count = -1;

	for (int i = 0; ; i++)
	{
		++round;
		std::cin >> x;
		if (x < 0)
			break;
		count[x]++;
		round_drop[x] = round;
		for (int i = 0; i < NUM; i++)
			if (count[i] > high_count)
				high_count = count[i];

		for (int i = 0; i < NUM; i++)
			if (count[i] == high_count)
				std::cout << i << " ";
		std::cout << '\n';
		for (int i = 0; i < NUM; i++) {
			if (round_drop[i] == 0 or (round - round_drop[i]) > k)
				std::cout << i << " ";
		}
		std::cout << '\n';

			if (x != 0) {
				if (x <= 10 || (x > 18 && x <= 28))
					if (x % 2 == 1)
						++reds;
					else              
						++blacks;
				else
					if (x % 2 == 0)
						++reds;
					else           
						++blacks;
				std::cout << reds << " " << blacks << '\n';
			}

		;
	}
}